service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/ClientInstallationServiceImpl.java

```diff
@@ -83,10 +83,17 @@ public void addInstallation(Variant variant, Installation entity) {
 
     private void mergeCategories(Installation entity) {
         if (entity.getCategories() != null) {
-            List<String> categoryNames = convertToNames(entity.getCategories());
-            final List<Category> categories = categoryDao.findByNames(categoryNames);
-            entity.getCategories().removeAll(categories);
-            entity.getCategories().addAll(categories);
+
+            if(!entity.getCategories().isEmpty()) {
+                final List<String> categoryNames = convertToNames(entity.getCategories());
+                final List<Category> categories = categoryDao.findByNames(categoryNames);
+                entity.getCategories().removeAll(categories);
+                entity.getCategories().addAll(categories);
```

Comment:
can you add the comment from above @sebastienblanc ? or do you want me to create a PR for it :)